src/components/structures/login/Login.js

```diff
@@ -102,6 +108,15 @@ module.exports = React.createClass({
     },
 
     onPasswordLogin: function(username, phoneCountry, phoneNumber, password) {
+        // Prevent people from submitting their password when homeserver
+        // discovery went wrong
+        if (this.state.discoveryError) return;
+
+        if (this.state.discoveredHsUrl) {
+            console.log("Rewriting username because the homeserver was discovered");
+            username = username.substring(1).split(":")[0];
```

Comment:
Should we not just be sending the full mxid to the server?

Reply:
this eventually gets passed along to `/login` as just the localpart. Here seemed like a good enough place to do the conversion is all.

Reply:
yea, around there. According to the spec ( https://matrix.org/docs/spec/client_server/r0.4.0.html#identifier-types ) it should work fine with the local part or user ID - I might have stumbled on a synapse bug (or I suck at testing)

Reply:
My testing was flawed - will fix.